dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/normalizer/SqlNormalizer.jj

```diff
@@ -0,0 +1,184 @@
+// Modified by SignalFx
+
+// This file is compiled with java -cp javacc.jar javacc SqlNormalizer.jj
+// I used javacc 7.0.5 the last time I ran it.
+// You might ask why I decided to simply check in the generated files rather than
+// integrate this step into the build system.
+// This is because the IDE usage gets awkward in that scenario without some build
+// adjustments which are complex; these can be revisited at any point in the future
+// Additionally note that I googleJavaFormat-ed the generated source
+options {
+	IGNORE_CASE = true;
+	UNICODE_INPUT = true;
+	STATIC = false;
+}
+
+PARSER_BEGIN(SqlNormalizer)
+
+package datadog.trace.instrumentation.jdbc.normalizer;
+
+import java.io.StringReader;
+
+public class SqlNormalizer {
+ public static String normalize(String s) throws ParseException {
+   SqlNormalizer parser = new SqlNormalizer(new StringReader(s));
+   return parser.Input().toString();
+ }
+}
+PARSER_END(SqlNormalizer)
+
+StringBuffer Input() :
+{
+	final int LIMIT = 32 * 1024;
+	StringBuffer sb = new StringBuffer();
+	Token t;
+}
+{
+	(
+	t = <Id>
+		{
+		sb.append(t.image);
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	| 
+	t = <BasicNum> 
+		{
+		sb.append('?');
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	| 
+	t = <HexNum> 
+		{
+		sb.append('?');
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	|
+	t = <QuotedStr> 
+		{
+		sb.append('?');
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	|
+	t = <DoubleQuotedStr> 
+		{
+		sb.append('?');
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	|
+	t = <Whitespace> 
+		{
+		sb.append(' ');
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	|
+	t = <Other> 
+		{
+		sb.append(t.image);
+		if (sb.length() > LIMIT) { return sb; }
+		}
+	)* <EOF>
+
+	{
+		return sb;
+	}
+}
+
+TOKEN :
+{
+// Javacc doesn't support unicode character categories like {Letter}, so here's my approximation of that.  No, I didn't type this by hand.
+
+	< Id: [
```

Comment:
I am not sure I understand what does this "Id" represent :) Can you add a comment, please?

Reply:
I could come up with a better name certainly - it's meant to represent a keyword (like SELECT) or identifier (like MY_SPECIAL_TABLE2).  Will change to KeywordOrIdentifier and a comment based on this explanation :-)